src/feature/control/control_getinfo.c

```diff
@@ -325,6 +325,51 @@ getinfo_helper_current_time(control_connection_t *control_conn,
   return 0;
 }
 
+/**
+ * Switch between microdesc vs networkstatus descriptor dumps
+ * Assumes that question would be either:
+ * - dir/status-vote/current/consensus-microdesc
+ * or
+ * - dir/status-vote/current/consensus
+ */
+STATIC int
+getinfo_helper_current_consensus(const char* question,
+                          char** answer,
+                          const char** errmsg)
+{
+  // Ensures question contains a request for
+  // ns or microdesc consensus
+  if (
+      strcmp(question, "dir/status-vote/current/consensus") &&
+      strcmp(question, "dir/status-vote/current/consensus-microdesc")) {
+    return 0;
+  }
+
+  const char* consensus_type = !strcmp(
+     question, "dir/status-vote/current/consensus-microdesc"
+  ) ? "microdesc" : "ns";
+
+  if (we_want_to_fetch_flavor(get_options(), FLAV_NS)) {
```

Comment:
Hmm, is this right? We call the `we_want_to_fetch_flavor()` with `FLAV_NS` but then we call `dirserv_get_consensus()` with arbitrary `consensus_type`?

Also, can you please document what's the purpose of this `we_want_to_fetch_flavor()` code block here?

Reply:
I think the we_want_to_fetch_flavor() check is used to make sure that the flavour is in RAM or on disk.